karapace/schema_registry_apis.py

```diff
@@ -56,20 +57,38 @@ class InvalidSchemaType(Exception):
 
 
 class KarapaceSchemaRegistry(KarapaceBase):
-    # pylint: disable=attribute-defined-outside-init
     def __init__(self, config: dict) -> None:
         super().__init__(config=config)
         self._add_schema_registry_routes()
-        self._init_schema_registry(config=config)
-
-    def _init_schema_registry(self, config: dict) -> None:  # pylint: disable=unused-argument
-        self.ksr = None
-        self.producer = None
         self.producer = self._create_producer()
-        self._create_master_coordinator()
-        self._create_schema_reader()
+        self.mc = MasterCoordinator(config=self.config)
+        self.mc.start()
+        self.ksr = KafkaSchemaReader(config=self.config, master_coordinator=self.mc)
+        self.ksr.start()
         self.schema_lock = asyncio.Lock()
 
+    def _create_producer(self) -> KafkaProducer:
+        while True:
+            try:
+                return KafkaProducer(
+                    bootstrap_servers=self.config["bootstrap_uri"],
+                    security_protocol=self.config["security_protocol"],
+                    ssl_cafile=self.config["ssl_cafile"],
+                    ssl_certfile=self.config["ssl_certfile"],
+                    ssl_keyfile=self.config["ssl_keyfile"],
+                    sasl_mechanism=self.config["sasl_mechanism"],
+                    sasl_plain_username=self.config["sasl_plain_username"],
+                    sasl_plain_password=self.config["sasl_plain_password"],
+                    api_version=(1, 0, 0),
+                    metadata_max_age_ms=self.config["metadata_max_age_ms"],
+                    max_block_ms=2000,  # missing topics will block unless we cache cluster metadata and pre-check
+                    connections_max_idle_ms=self.config["connections_max_idle_ms"],  # helps through cluster upgrades ??
+                    kafka_client=KarapaceKafkaClient,
+                )
+            except:  # pylint: disable=bare-except
```

Comment:
This one I didn't dig to see what is actually raised, will have to wait. This code is moved from the base.py